modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -0,0 +1,676 @@
+                                         , in_constants
+                                         , m_gim.metadata(nh).get<IslandExec>().object});
+
+                // Initialize queues for every operation's input
+                ade::TypedGraph<DataQueue> qgr(*m_island_graph);
+                for (auto eh : nh->inEdges())
+                {
+                    // ...only if the data is not compile-const
+                    if (const_ins.count(eh->srcNode()) == 0) {
+                        qgr.metadata(eh).set(DataQueue(queue_capacity));
+                    }
+                }
+            }
+            break;
+        case NodeKind::SLOT:
+            {
+                const auto orig_data_nh
+                    = m_gim.metadata(nh).get<DataSlot>().original_data_node;
+                m_slots.emplace_back(DataDesc{nh, orig_data_nh});
+            }
+            break;
+        case NodeKind::EMIT:
+            {
+                const auto emitter_idx
+                    = m_gim.metadata(nh).get<Emitter>().proto_index;
+                GAPI_Assert(emitter_idx < m_emitters.size());
+                m_emitters[emitter_idx] = nh;
+            }
+            break;
+        case NodeKind::SINK:
+            {
+                const auto sink_idx
+                    = m_gim.metadata(nh).get<Sink>().proto_index;
+                GAPI_Assert(sink_idx < m_sinks.size());
+                m_sinks[sink_idx] = nh;
+
+                // Also initialize Sink's input queue
+                ade::TypedGraph<DataQueue> qgr(*m_island_graph);
+                GAPI_Assert(nh->inEdges().size() == 1u);
+                qgr.metadata(nh->inEdges().front()).set(DataQueue(queue_capacity));
+                m_sink_queues[sink_idx] = &qgr.metadata(nh->inEdges().front()).get<DataQueue>().q;
+            }
+            break;
+        default:
+            GAPI_Assert(false);
+            break;
+        } // switch(kind)
+    } // for(gim nodes)
+    m_out_queue.set_capacity(queue_capacity);
+}
+
+void cv::gimpl::GStreamingExecutor::setSource(GRunArgs &&ins)
+{
+    bool has_video = false;
+
+    // Walk through the protocol, set-up emitters appropriately
+    // There's a 1:1 mapping between emitters and corresponding data inputs.
+    for (auto it : ade::util::zip(ade::util::toRange(m_emitters),
+                                  ade::util::toRange(ins)))
+    {
+        auto  emit_nh  = std::get<0>(it);
+        auto& emit_arg = std::get<1>(it);
+        auto& emitter  = m_gim.metadata(emit_nh).get<Emitter>().object;
+
+        using T = GRunArg;
+        switch (emit_arg.index())
+        {
+        // Create a streaming emitter.
+        // Produces the next video frame when pulled.
+        case T::index_of<cv::gapi::GVideoCapture>():
+            if (has_video)
+                util::throw_error(std::logic_error("Only one video source is"
+                                                   " currently supported!"));
+            has_video = true;
```

Comment:
...maybe it is more clear to check this separately (like `count_if` or `find_if`)

Reply:
Done